src/script_opt/ZAM/BuiltInSupport.h

```diff
@@ -36,12 +36,12 @@ class CatArg {
 
 class FixedCatArg : public CatArg {
 public:
-    FixedCatArg(const TypePtr& t);
+    FixedCatArg(TypePtr t);
 
     void RenderInto(ZVal* zframe, int slot, char*& res) override;
 
 protected:
-    const TypePtr& t;
+    TypePtr t;
```

Comment:
Seems you could continue to use `FixedCatArg(const TypePtr& t)` as that will be copied, but either works.

Reply:
I like it being a `TypePtr` in the constructor declaration because that way it's more apparent that the object is going to hang onto the pointer. `const TypePtr&` is more suggestive that it just uses it during construction but not later. (Not that the caller should actually care about this distinction, though ...)

Reply:
A yes, I like that, thanks!